core/src/Kokkos_ScratchSpace.hpp

```diff
@@ -31,103 +31,71 @@ static_assert(false,
 
 namespace Kokkos {
 
-/** \brief  Scratch memory space associated with an execution space.
- *
- */
-template <class ExecSpace>
-class ScratchMemorySpace {
+template <typename ExecSpace, typename PointerType>
```

Comment:
We could probably do without a `PointerType` template parameter. It does really only matter what we are doing when accessing the data.